lib/hammer_cli_foreman/compute_profile.rb

```diff
@@ -0,0 +1,422 @@
+             :required => true,
+             :format => HammerCLI::Options::Normalizers::Number.new
+      extend_help do |h|
+        h.section _('Provider Interface specific options') do |h|
+          ::HammerCLIForeman::ComputeProfile.compute_resources.each do |name, provider|
+            h.section name do |h|
+              if defined?(provider.interface_create_help)
+                provider.interface_create_help(h)
+              end
+            end
+          end
+        end
+      end
+
+      def validate_options
+        super
+        validator.any(:option_compute_profile_id, :option_compute_profile_name).required
+        validator.any(:option_compute_resource_id, :option_compute_resource_name).required
+        validator.all(:option_interface_id, :option_set_values).required
+      end
+
+      def request_params
+        params = HammerCLIForeman::ComputeProfile.get_params(options)
+        interface_attr =  params['compute_attribute']['vm_attrs']['interfaces_attributes'] || {}
+        params['id'] = params['compute_attribute']['id']
+        params['compute_attribute']['vm_attrs']['interfaces_attributes'] = interface_attr.merge({option_interface_id => option_set_values})
+        params
+      end
+      success_message _('Interface was updated.')
+      failure_message _('Could not update interface')
+      build_options :without => :id
+    end
+
+    class RemoveInterface < HammerCLIForeman::UpdateCommand
+      command_name _('remove-interface')
+      resource :compute_attributes
+      desc _('Remove compute profile interface')
+      option '--interface-id', 'INTERFACE ID', _('Interface id'),
+             :format => HammerCLI::Options::Normalizers::Number.new
+
+      def validate_options
+        super
+        validator.any(:option_compute_profile_id, :option_compute_profile_name).required
+        validator.any(:option_compute_resource_id, :option_compute_resource_name).required
+        validator.all(:option_interface_id).required
+      end
+
+      def request_params
+        params = HammerCLIForeman::ComputeProfile.get_params(options)
+        params['id'] = params['compute_attribute']['id']
+        if params['compute_attribute']['vm_attrs']['interfaces_attributes'].has_key?(option_interface_id.to_s)
+          params['compute_attribute']['vm_attrs']['interfaces_attributes'].delete(option_interface_id.to_s)
+        else
+          signal_usage_error _('unknown interface id')
+        end
+        params
+      end
+      success_message _('Interface was removed.')
+      failure_message _('Could not remove interface')
+      build_options :without => :id
+    end
+
+    class AddVolume < HammerCLIForeman::UpdateCommand
+      command_name _('add-volume')
+      resource :compute_attributes
+      desc _('Add volume for Compute Profile')
+
+      option '--set-values', 'VOLUME', _('Volume parameters, should be comma separated list of values.'),
```

Comment:
Sorry for bothering you with this again, but there is a convention, that says option descriptions should not end with a dot. Simply remove '.' at the end of option description.